src/FileHelper.php

```diff
@@ -143,21 +143,55 @@ public static function normalizePath(string $path): string
      * Removes a directory (and all its content) recursively.
      *
      * @param string $directory the directory to be deleted recursively.
-     * @param array $options options for directory remove. Valid options are:
+     * @param array $options options for directory remove ({@see clearDirectory()}).
+     *
+     * @return void
+     */
+    public static function removeDirectory(string $directory, array $options = []): void
+    {
+        try {
+            $handle = static::openDirectory($directory);
+        } catch (\InvalidArgumentException $e) {
+            return;
+        }
+
+        static::clearDirectoryByHandle($handle, $directory, $options);
+
+        if (is_link($directory)) {
+            self::unlink($directory);
+        } else {
+            rmdir($directory);
+        }
+    }
+
+    /**
+     * Clear all directory content.
+     *
+     * @param string $directory the directory to be cleared.
+     * @param array $options options for directory clear . Valid options are:
      *
      * - traverseSymlinks: boolean, whether symlinks to the directories should be traversed too.
      *   Defaults to `false`, meaning the content of the symlinked directory would not be deleted.
      *   Only symlink would be removed in that default case.
      *
      * @return void
      */
-    public static function removeDirectory(string $directory, array $options = []): void
+    public static function clearDirectory(string $directory, array $options = []): void
     {
-        if (!empty($options['traverseSymlinks']) || !is_link($directory)) {
-            if (!($handle = @opendir($directory))) {
-                return;
-            }
+        $handle = static::openDirectory($directory);
+        static::clearDirectoryByHandle($handle, $directory, $options);
```

Comment:
There's no need for `clearDirectoryByHandle`. Code could be moved into this method.